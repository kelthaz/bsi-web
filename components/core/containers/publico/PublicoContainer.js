import Footer from '../../footer/Footer';
import Header from '../../header/Header';

const PublicoContainer = ({ pageComponent, servicesData }) => {
  const { component: Component } = pageComponent;

  return (
    <>
      <Header />
      <Component {...servicesData} />
      <Footer />
    </>
  );
};

export default PublicoContainer;
