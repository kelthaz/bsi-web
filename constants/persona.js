export const MORAL = 'MORAL';
export const FISICA = 'FISICA';
