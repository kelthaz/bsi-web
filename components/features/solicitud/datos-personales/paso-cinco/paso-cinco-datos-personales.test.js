import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { useRouter } from 'next/router';
import { act } from 'react-dom/test-utils';
import storeTest from '../../../../../redux/storeTest';
import PasoCincoDatosPersonales from './PasoCincoDatosPersonales';
import {
  AGRADECIMIENTO_DATOS_PERSONA_ROUTE,
  PASO_CINCO_DATOS_PERSONA_ROUTE,
  PASO_CUATRO_DATOS_PERSONA_ROUTE,
} from '../../../../../constants/routes/solicitud/persona';
import LoginRepositorio from '../../../../../services/login/login.repositorio';
import { regexHyphen } from '../../../../../constants/regex';
import { MORAL } from '../../../../../constants/persona';
import { AVISO_ROUTE } from '../../../../../constants/routes/publico/publico';
import { rfcInvalido } from '../../../../../constants/errors';
import ModalActualizar from '../../../../core/containers/solicitud/modal-actualizar/ModalActualizar';
import TabInformativo from '../../../../core/containers/solicitud/tab-informativo/TabInformativo';
import ListaNegraRepositorio from '../../../../../services/solicitud/listaNegra.repositorio';

jest.mock('../../../../../services/login/login.repositorio');
jest.mock('../../../../../services/solicitud/listaNegra.repositorio');

describe('Pruebas en el componente PasoCincoDatosPersonales', () => {
  const push = jest.fn();
  const beforePopState = jest.fn();
  const datoPersona = {
    primerNombre: 'JESUS',
    segundoNombre: 'D',
    primerApellido: 'DAVID',
    segundoApellido: '',
    tipoPersona: 'FISICA',
    razonSocial: 'Paletas',
    tipoSociedad: {
      value: 10,
      label: 'S.A.',
    },
    nombreEmpresa: 'PALETITAS',
    rfcRepresentante: 'JUPU800825561',
    sector: {
      value: 1,
      label: 'SERVICIOS PROFESIONALES Y TÉCNICO',
    },
    giro: {
      value: 1,
      label: 'ARRENDAMIENTO DE INMUEBLES COMERCIALES E INDUSTRIALES.',
    },
    descripcionEmpresa: 'vendo paletas',
    celular: '22-2222-2222',
    correo: 'paletas@gmail.com',
  };
  useRouter.mockImplementation(() => ({ push, beforePopState }));
  // prefetch.mockImplementation(() => {});

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', async () => {
    const store = storeTest();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test('Debe de existir el link con el href hacia aviso', async () => {
    // arrange
    const store = storeTest();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );
    // act
    const link = wrapper.find({ href: AVISO_ROUTE });
    // assert
    expect(link.exists()).toBe(true);
    wrapper.unmount();
  });

  test('Debe de guardar en el reducer rfc, contrasena, confirmarContrasena, aceptoTerminos y enviarlo al paso que selecciono, como persona Fisica', async () => {
    // arrange
    const store = storeTest();

    const wrapper = mount(
      <Provider store={store}>
        <TabInformativo
          tabs={[{ path: 'documentacion', label: 'Documentación' }]}
          currentTab="datos-personales"
          currentStep={6}
          valipStep={5}
          steps={[
            { step: 4, paso: 5, tipoPersona: '', action: PASO_CUATRO_DATOS_PERSONA_ROUTE },
            { step: 5, paso: 6, tipoPersona: '', action: PASO_CINCO_DATOS_PERSONA_ROUTE },
          ]}
        />
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    const rfc = wrapper.find('input').find({ name: 'rfc' });
    const contrasena = wrapper.find('input').find({ name: 'contrasena' });
    const confirmarContrasena = wrapper.find('input').find({ name: 'confirmarContrasena' });
    const aceptoTerminos = wrapper.find('input').find({ name: 'aceptoTerminos' });
    // act
    await act(async () => {
      rfc.simulate('change', {
        target: { value: 'CUPU800825569', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      contrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      confirmarContrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      aceptoTerminos.simulate('change');
    });

    await act(async () => {
      wrapper.find('button.buttons-arrow').at(0).simulate('click');
    });

    wrapper.update();

    await act(async () => {
      wrapper.find('.btn-medium-secondary').simulate('click');
    });
    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // assert
    expect(datosPersonales.rfc).toEqual(wrapper.find('input').find({ name: 'rfc' }).prop('value'));
    expect(datosPersonales.contrasena).toEqual(wrapper.find('input').find({ name: 'contrasena' }).prop('value'));
    expect(datosPersonales.confirmarContrasena).toEqual(
      wrapper.find('input').find({ name: 'confirmarContrasena' }).prop('value')
    );
    expect(datosPersonales.aceptoTerminos).toEqual(
      wrapper.find('input').find({ name: 'aceptoTerminos' }).prop('checked')
    );

    expect(store.getState().solicitud.currentStep.lastStep).toBe(true);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_CUATRO_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de llamar el servicio de registro y guardar la data del reducer y enviarlo al paso de agradecimiento, como persona Moral', async () => {
    // arrange
    const store = storeTest();
    const { solicitud, simulador } = store.getState();
    solicitud.datosPersonales = {
      ...solicitud.datosPersonales,
      ...datoPersona,
      tipoPersona: MORAL,
    };
    simulador.dataSimulador.plazo = { label: '12 meses' };
    simulador.dataSimulador.periodicidad = { label: 'Mensuales' };

    const wrapper = mount(
      <Provider store={store}>
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    await LoginRepositorio.postRegistro.mockResolvedValue();
    await ListaNegraRepositorio.postListaNegra.mockResolvedValue({ data: { resultadoConsulta: '0' } });

    const form = wrapper.find('form');
    const rfc = wrapper.find('input').find({ name: 'rfc' });
    const contrasena = wrapper.find('input').find({ name: 'contrasena' });
    const confirmarContrasena = wrapper.find('input').find({ name: 'confirmarContrasena' });
    const aceptoTerminos = wrapper.find('input').find({ name: 'aceptoTerminos' });
    // act
    await act(async () => {
      rfc.simulate('change', {
        target: { value: 'CUPU800825569', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      contrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      confirmarContrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      aceptoTerminos.simulate('change');
    });

    wrapper.update();

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales, sincronizado },
      simulador: {
        dataSimulador: { monto, plazo, periodicidad },
      },
    } = store.getState();
    const data = {
      ...datosPersonales,
      celular: datosPersonales.celular.replace(regexHyphen, ''),
      tipoSociedad: datosPersonales.tipoSociedad?.value,
      sector: datosPersonales.sector.value,
      giro: datosPersonales.giro.value,
      rfc: wrapper.find('input').find({ name: 'rfc' }).prop('value'),
      password: wrapper.find('input').find({ name: 'contrasena' }).prop('value'),
      simulador: { monto, plazo: plazo.value, periodicidad: periodicidad.value },
    };
    delete data.aceptoTerminos;
    delete data.contrasena;
    delete data.confirmarContrasena;
    delete data.enListaNegra;

    // assert
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledTimes(1);
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledWith(data);
    expect(sincronizado).toBe(false);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(AGRADECIMIENTO_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de llamar el servicio de registro y guardar la data del reducer y enviarlo al paso de agradecimiento, como persona Fisica', async () => {
    // arrange
    const store = storeTest();
    const { solicitud, simulador } = store.getState();
    solicitud.datosPersonales = {
      ...solicitud.datosPersonales,
      ...datoPersona,
    };
    simulador.dataSimulador.plazo = { label: '12 meses' };
    simulador.dataSimulador.periodicidad = { label: 'Mensuales' };

    const wrapper = mount(
      <Provider store={store}>
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    await ListaNegraRepositorio.postListaNegra.mockResolvedValue({ data: { resultadoConsulta: '0' } });
    await LoginRepositorio.postRegistro.mockResolvedValue();

    const form = wrapper.find('form');
    const rfc = wrapper.find('input').find({ name: 'rfc' });
    const contrasena = wrapper.find('input').find({ name: 'contrasena' });
    const confirmarContrasena = wrapper.find('input').find({ name: 'confirmarContrasena' });
    const aceptoTerminos = wrapper.find('input').find({ name: 'aceptoTerminos' });
    // act
    await act(async () => {
      rfc.simulate('change', {
        target: { value: 'CUPU800825569', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      contrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      confirmarContrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      aceptoTerminos.simulate('change');
    });

    wrapper.update();

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales, sincronizado },
      simulador: {
        dataSimulador: { monto, plazo, periodicidad },
      },
    } = store.getState();
    const data = {
      ...datosPersonales,
      celular: datosPersonales.celular.replace(regexHyphen, ''),
      tipoSociedad: datosPersonales.tipoSociedad?.label,
      sector: datosPersonales.sector.value,
      giro: datosPersonales.giro.value,
      rfc: wrapper.find('input').find({ name: 'rfc' }).prop('value'),
      password: wrapper.find('input').find({ name: 'contrasena' }).prop('value'),
      simulador: { monto, plazo: plazo.value, periodicidad: periodicidad.value },
    };
    delete data.aceptoTerminos;
    delete data.contrasena;
    delete data.confirmarContrasena;
    delete data.tipoSociedad;
    delete data.razonSocial;
    delete data.enListaNegra;
    delete data.rfcRepresentante;

    // assert
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledTimes(1);
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledWith(data);
    expect(sincronizado).toBe(false);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(AGRADECIMIENTO_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de llamar el servicio de registro y fallar, como persona Fisica', async () => {
    // arrange
    const store = storeTest();
    const { solicitud, simulador } = store.getState();
    solicitud.datosPersonales = {
      ...solicitud.datosPersonales,
      ...datoPersona,
    };
    simulador.dataSimulador.plazo = { label: '12 meses' };
    simulador.dataSimulador.periodicidad = { label: 'Mensuales' };

    const wrapper = mount(
      <Provider store={store}>
        <PasoCincoDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    await ListaNegraRepositorio.postListaNegra.mockResolvedValue({ data: { resultadoConsulta: '0' } });
    await LoginRepositorio.postRegistro.mockRejectedValue();

    const form = wrapper.find('form');
    const rfc = wrapper.find('input').find({ name: 'rfc' });
    const contrasena = wrapper.find('input').find({ name: 'contrasena' });
    const confirmarContrasena = wrapper.find('input').find({ name: 'confirmarContrasena' });
    const aceptoTerminos = wrapper.find('input').find({ name: 'aceptoTerminos' });
    // act
    await act(async () => {
      rfc.simulate('change', {
        target: { value: 'CUPU800825569', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      contrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      confirmarContrasena.simulate('change', {
        target: { value: 'abcdefA6', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      aceptoTerminos.simulate('change');
    });

    wrapper.update();

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales, currentStep },
      simulador: {
        dataSimulador: { monto, plazo, periodicidad },
      },
    } = store.getState();
    const data = {
      ...datosPersonales,
      celular: datosPersonales.celular.replace(regexHyphen, ''),
      tipoSociedad: datosPersonales.tipoSociedad?.label,
      sector: datosPersonales.sector.value,
      giro: datosPersonales.giro.value,
      rfc: wrapper.find('input').find({ name: 'rfc' }).prop('value'),
      password: wrapper.find('input').find({ name: 'contrasena' }).prop('value'),
      simulador: { monto, plazo: plazo.value, periodicidad: periodicidad.value },
    };
    delete data.aceptoTerminos;
    delete data.contrasena;
    delete data.confirmarContrasena;
    delete data.tipoSociedad;
    delete data.razonSocial;
    delete data.enListaNegra;
    delete data.rfcRepresentante;
    // assert
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledTimes(1);
    expect(LoginRepositorio.postRegistro).toHaveBeenCalledWith(data);
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep);
    expect(push).toHaveBeenCalledTimes(0);
    wrapper.unmount();
  });

  test('Debe aparecer error en el campo del rfc debido a que se cambio el tipo de persona', async () => {
    // arrange
    const store = storeTest();
    const { solicitud } = store.getState();
    solicitud.datosPersonales = {
      rfc: 'CUPU800825569',
      contrasena: 'aaaaaaA6',
      confirmarContrasena: 'aaaaaaA6',
      aceptoTerminos: true,
      tipoPersona: MORAL,
    };
    solicitud.currentStep = {
      ...solicitud.currentStep,
      lastStep: true,
    };

    let wrapper;
    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <PasoCincoDatosPersonales validate />
          <ModalActualizar />
        </Provider>
      );
    });

    const rfc = wrapper.find('TextField').find({ name: 'rfc' });
    // act
    const error = rfc.find('span').text();
    // assert
    expect(error).toBe(rfcInvalido());
    wrapper.unmount();
  });
});
