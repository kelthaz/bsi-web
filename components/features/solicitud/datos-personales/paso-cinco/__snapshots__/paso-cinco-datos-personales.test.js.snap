// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PasoCincoDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PasoCincoDatosPersonales
    validate={true}
  >
    <div
      className="contedor-fixed-tab"
    >
      <div
        className="contedor-solicitud"
      >
        <div
          className="container p-0"
        >
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <h2
              className="color-blue-storm"
            >
              ¡Sólo falta crear tu cuenta!
            </h2>
            <p
              className="color-dark-gray sub"
            >
              Tu usuario será el RFC con el que facturas y crearás una contraseña. Con tu cuenta podrás retomar el proceso en cualquier momento.
            </p>
            <div
              className="row no-gutters"
            >
              <div
                className="col-lg-3 col-md-4 col-sm-12 col-xs-12 "
              >
                <p
                  className="input color-gray"
                >
                  Mi RFC es
                </p>
              </div>
              <div
                className="col-lg-6 col-md-6  col-xs-12 pr-lg-2 pr-md-2 pb-sm-3 pb-xs-3"
              >
                <TextField
                  afterBlur={[Function]}
                  capitalize={false}
                  disabled={false}
                  format="rfcformatter"
                  formulario={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "getFieldHelpers": [Function],
                      "getFieldMeta": [Function],
                      "getFieldProps": [Function],
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialErrors": Object {},
                      "initialStatus": undefined,
                      "initialTouched": Object {},
                      "initialValues": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                      "isSubmitting": false,
                      "isValid": true,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validateOnMount": false,
                      "values": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                    }
                  }
                  inverted={false}
                  label="TLMF160693H17"
                  maxlength={13}
                  name="rfc"
                  optional={false}
                  paste={true}
                  readonly={false}
                  size="big"
                  type="text"
                  validation={false}
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  false "
                      disabled={false}
                      id="rfc"
                      maxLength={13}
                      name="rfc"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="TLMF160693H17"
                      readOnly={false}
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <span
                      className="help-text"
                    >
                       
                    </span>
                  </div>
                </TextField>
              </div>
              <div
                className="col-lg-5 col-md-6 col-sm-12 col-xs-12 "
              >
                <p
                  className="input color-gray"
                >
                  Mi contraseña será
                </p>
              </div>
              <div
                className="col-lg-5 col-md-5 col-xs-12"
              >
                <TextField
                  afterBlur={[Function]}
                  capitalize={false}
                  disabled={false}
                  format="passwordspace"
                  formulario={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "getFieldHelpers": [Function],
                      "getFieldMeta": [Function],
                      "getFieldProps": [Function],
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialErrors": Object {},
                      "initialStatus": undefined,
                      "initialTouched": Object {},
                      "initialValues": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                      "isSubmitting": false,
                      "isValid": true,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validateOnMount": false,
                      "values": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                    }
                  }
                  inverted={false}
                  label="Contraseña"
                  maxlength={20}
                  name="contrasena"
                  optional={false}
                  paste={false}
                  readonly={false}
                  size="big"
                  type="password"
                  validation={false}
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big-password  false "
                      disabled={false}
                      id="contrasena"
                      maxLength={20}
                      name="contrasena"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Contraseña"
                      readOnly={false}
                      tabIndex="0"
                      type="password"
                      value=""
                    />
                    <span
                      className="help-text"
                    >
                       
                    </span>
                    <button
                      className="button-password-inverted"
                      onClick={[Function]}
                      type="button"
                    >
                      <SvgHidenPassword>
                        <svg
                          fill="none"
                          height={24}
                          viewBox="0 0 24 24"
                          width={24}
                        >
                          <path
                            d="M22 9l-1.8 2.159A10.675 10.675 0 0112 15v0a10.675 10.675 0 01-8.2-3.841L2 9"
                            stroke="gray"
                            strokeLinecap="round"
                            strokeWidth={2}
                          />
                          <path
                            d="M1.2 15.4a1 1 0 001.6 1.2l-1.6-1.2zm1.6 1.2l3-4-1.6-1.2-3 4 1.6 1.2zM22.8 15.4a1 1 0 01-1.6 1.2l1.6-1.2zm-1.6 1.2l-3-4 1.6-1.2 3 4-1.6 1.2zM7.03 18.758a1 1 0 101.94.485l-1.94-.485zm1.94.485l1-4-1.94-.486-1 4 1.94.485zM16.97 18.758a1 1 0 11-1.94.485l1.94-.485zm-1.94.485l-1-4 1.94-.486 1 4-1.94.485z"
                            fill="gray"
                          />
                        </svg>
                      </SvgHidenPassword>
                    </button>
                  </div>
                </TextField>
              </div>
              <div
                className="col-lg-8 col-md-11 "
              >
                <ValidatePassword
                  field="contrasena"
                  formulario={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "getFieldHelpers": [Function],
                      "getFieldMeta": [Function],
                      "getFieldProps": [Function],
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialErrors": Object {},
                      "initialStatus": undefined,
                      "initialTouched": Object {},
                      "initialValues": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                      "isSubmitting": false,
                      "isValid": true,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validateOnMount": false,
                      "values": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                    }
                  }
                />
              </div>
              <div
                className="col-lg-5 col-md-6 col-xs-12 "
              >
                <p
                  className="input color-gray"
                >
                  Confirmo que será
                </p>
              </div>
              <div
                className="col-lg-5 col-md-5 col-xs-12 pb-sm-3 pb-xs-3"
              >
                <TextField
                  afterBlur={[Function]}
                  capitalize={false}
                  disabled={false}
                  format="passwordspace"
                  formulario={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "getFieldHelpers": [Function],
                      "getFieldMeta": [Function],
                      "getFieldProps": [Function],
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialErrors": Object {},
                      "initialStatus": undefined,
                      "initialTouched": Object {},
                      "initialValues": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                      "isSubmitting": false,
                      "isValid": true,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validateOnMount": false,
                      "values": Object {
                        "aceptoTerminos": null,
                        "confirmarContrasena": "",
                        "contrasena": "",
                        "rfc": "",
                      },
                    }
                  }
                  inverted={false}
                  label="Contraseña"
                  maxlength={20}
                  name="confirmarContrasena"
                  optional={false}
                  paste={false}
                  readonly={false}
                  size="big"
                  type="password"
                  validation={false}
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big-password  false "
                      disabled={false}
                      id="confirmarContrasena"
                      maxLength={20}
                      name="confirmarContrasena"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Contraseña"
                      readOnly={false}
                      tabIndex="0"
                      type="password"
                      value=""
                    />
                    <span
                      className="help-text"
                    >
                       
                    </span>
                    <button
                      className="button-password-inverted"
                      onClick={[Function]}
                      type="button"
                    >
                      <SvgHidenPassword>
                        <svg
                          fill="none"
                          height={24}
                          viewBox="0 0 24 24"
                          width={24}
                        >
                          <path
                            d="M22 9l-1.8 2.159A10.675 10.675 0 0112 15v0a10.675 10.675 0 01-8.2-3.841L2 9"
                            stroke="gray"
                            strokeLinecap="round"
                            strokeWidth={2}
                          />
                          <path
                            d="M1.2 15.4a1 1 0 001.6 1.2l-1.6-1.2zm1.6 1.2l3-4-1.6-1.2-3 4 1.6 1.2zM22.8 15.4a1 1 0 01-1.6 1.2l1.6-1.2zm-1.6 1.2l-3-4 1.6-1.2 3 4-1.6 1.2zM7.03 18.758a1 1 0 101.94.485l-1.94-.485zm1.94.485l1-4-1.94-.486-1 4 1.94.485zM16.97 18.758a1 1 0 11-1.94.485l1.94-.485zm-1.94.485l-1-4 1.94-.486 1 4-1.94.485z"
                            fill="gray"
                          />
                        </svg>
                      </SvgHidenPassword>
                    </button>
                  </div>
                </TextField>
              </div>
            </div>
            <div
              className="row no-gutters"
            >
              <CheckTextBox
                formulario={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "getFieldHelpers": [Function],
                    "getFieldMeta": [Function],
                    "getFieldProps": [Function],
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialErrors": Object {},
                    "initialStatus": undefined,
                    "initialTouched": Object {},
                    "initialValues": Object {
                      "aceptoTerminos": null,
                      "confirmarContrasena": "",
                      "contrasena": "",
                      "rfc": "",
                    },
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "status": undefined,
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validateOnMount": false,
                    "values": Object {
                      "aceptoTerminos": null,
                      "confirmarContrasena": "",
                      "contrasena": "",
                      "rfc": "",
                    },
                  }
                }
                name="aceptoTerminos"
              >
                <div
                  className="card-simple-blue-light container-check-text"
                >
                  <div
                    className="container-check"
                  >
                    <input
                      checked={false}
                      className=" my-check"
                      id="aceptoTerminos"
                      name="aceptoTerminos"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <label
                      className="label"
                      htmlFor="aceptoTerminos"
                    >
                       
                    </label>
                  </div>
                  <div>
                    <p
                      className="m-0"
                    >
                      <span>
                        Acepto: (1) los 
                      </span>
                      <a
                        className="link"
                        rel="noreferrer"
                        target="_blank"
                      >
                        Términos y Condiciones
                      </a>
                      <span>
                        , (2) el 
                      </span>
                      <Component
                        href="/aviso-privacidad"
                      >
                        <a
                          className="link"
                        >
                          Aviso de Privacidad
                        </a>
                      </Component>
                      <span>
                        , (3) tu Solicitud de Crédito y que (4) los productos y/o servicios que ofrece BanCoppel serán promocionados, aceptados y/o modificados a través de medios electrónicos, telefónicos, digitales y/o cualquier otra tecnología.
                      </span>
                    </p>
                  </div>
                </div>
                <span
                  className="color-red"
                >
                   
                </span>
              </CheckTextBox>
            </div>
            <div
              className="flex-column-center-config my-3 "
            >
              <button
                aria-label="Avanzar"
                className="btn-medium"
                disabled={true}
                type="submit"
              >
                <span>
                  Crea tu contraseña
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </PasoCincoDatosPersonales>
  <ModalActualizar>
    <Modal
      closeable={true}
      onClose={[Function]}
      openModal={false}
      setOpenModal={[Function]}
    />
  </ModalActualizar>
</Provider>
`;
